go/arrow/array/struct.go

```diff
@@ -105,6 +107,36 @@ func (a *Struct) setData(data *Data) {
 	}
 }
 
+func (a *Struct) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+
+	tmp := make(map[string]interface{})
+	fieldList := a.data.dtype.(*arrow.StructType).Fields()
+	for j, d := range a.fields {
+		tmp[fieldList[j].Name] = d.getOneForMarshal(i)
+	}
+	return tmp
+}
+
+func (a *Struct) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+
+	buf.WriteByte('[')
+	for i := 0; i < a.Len(); i++ {
```

Comment:
I might be getting a little lost here but IIUC the other json functions return a list representing the column value.  Here it seems we transpose each column to a row first?  This seems to lack symmetry?  Maybe in this case the MarshalJSON function shouldn't use getOneForMarshal?

Reply:
In a struct column, each "value" in the list, is an object consisting of keys from the struct field names with values for the corresponding index. 

For example, a `struct<a: int, b: float>` column would be marshalled to `[ {"a": <a[0]>, "b": <b[0]>}, {"a": <a[1]>, "b": <b[1]> }, .... ]`. Does that make more sense?

Reply:
Yeah, so I think I've come around to how this is implemented.  I think what I was struggling with is an IntArray, gets serialized as `[1,2,3]` but one could maybe think it should get serialized as something like `[[1],[2],[3]` or `[{'entry':1}, {'entry':2}, {'entry': 3]` similar to what struct does).  I guess it is good if someone just wants to serialize/deserialize just an array and not a record batch that it happens as a flat JSON list and the row by row approach is taken for struct.